shell/platform/fuchsia/dart_runner/dart_test_component_controller_v2.h

```diff
@@ -68,7 +69,9 @@ class DartTestComponentControllerV2
  private:
   /// Helper for actually running the Dart main. Returns true if successful,
   /// false otherwise.
-  bool RunDartMain();
+  fpromise::promise<> RunDartMain();
+
+  void RunPromiseToCompletion(fpromise::promise<> promise);
```

Comment:
We've written this like it will take any promise and run it to completion, but it is designed around a very specific promise (the RunDartMain promise) since it relies on `has_set_return_code`.

Instead of abstracting this into a RunPromiseToCompletion method, would it be better to inline the code, i.e. replace

```c++
    RunPromiseToCompletion(std::move(dart_main_promise));
```

with

```c++
  auto dart_state = tonic::DartState::Current();
  executor_.schedule_task(std::move(dart_main_promise));
  // looping logic ...
```

and remove this method?